### PR TITLE
fix(use-click-away): update handler reference

### PR DESCRIPTION
**Pull request by tmkx**
## Checklist

- [x] Fix linting errors
- [x] Tests have been added / updated (or snapshots)

## Change information

1. The `handler` should be updated if changed.
2. replace `Event` with `MouseEvent`

### CODE REVIEW

File: components/use-click-away/__tests__/click-away.test.tsx [hunk 1]
```diff
@@ -34,4 +34,23 @@ describe('UseClickAway', () => {
...
+      const handler = isFirstRender ? handler1 : handler2
+      useClickAway(ref, handler)
+      isFirstRender = false
+    })
+    simulateNativeClick(document.body)
+    expect(handler1).toHaveBeenCalled()
+
+    rerender()
+    simulateNativeClick(document.body)
+    expect(handler2).toHaveBeenCalled()
```
Comment (unix) [inline components/use-click-away/__tests__/click-away.test.tsx:54]: `handler1` should not be called?

**Review by unix — APPROVED** [review]
Great work 👍

**Comment by unix** [comment]
Thanks for your contribution, if you have no other changes we can just merge?
And, do we need to publish a patch version?(if you need it)

File: components/use-click-away/__tests__/click-away.test.tsx (hunk 1 above)
Comment (tmkx) [reply components/use-click-away/__tests__/click-away.test.tsx:54]: `handler1` and `handler2` should be called once each in this test.

Comment (tmkx) [reply components/use-click-away/__tests__/click-away.test.tsx:54]: This assertion is mainly ensure `handler2` has been called.

**Comment by tmkx** [comment]
> And, do we need to publish a patch version? @tmkx (if you need it)

Just merge, no patch version is needed, thx.

**Pull request merged**